.github/workflows/tests-clustermesh-upgrade.yaml

```diff
@@ -748,17 +748,40 @@ jobs:
     if: ${{ always() }}
     name: Merge and Upload Artifacts
     runs-on: ubuntu-latest
-    needs: upgrade-and-downgrade
     steps:
+      # TODO remove list and filter artifact steps if this PR https://github.com/actions/upload-artifact/pull/521 is merged
```

Comment:
This might be nice but I am unsure if upstream PR is merged and the action ignores non-existent artifacts.
Maybe, we can do this in another PR?

Reply:
I don't personally have much hope in the upstream PR, considering that the last interaction seems to date back in march. My main concern with this replicated boilerplate code is that it can easily get out of sync across workflows, leading to subtle differences. However, I don't see that as a hard blocker, it could be a follow-up as well.